modules/button/react/lib/Button.tsx

```diff
@@ -41,27 +41,27 @@ export interface ButtonProps<T = ButtonTypes | BetaButtonTypes>
 }
 
 export default class Button extends React.Component<ButtonProps> {
-  public static Types = ButtonTypes;
-  public static Sizes = ButtonSizes;
+  public static Variant = ButtonVariant;
+  public static Size = ButtonSize;
 
   static defaultProps = {
-    buttonSize: ButtonSizes.Large,
-    buttonType: ButtonTypes.Secondary,
+    size: ButtonSize.Large,
+    variant: ButtonVariant.Secondary,
     grow: false,
   };
 
   public render() {
-    const {buttonRef, ...elemProps} = this.props;
+    const {variant, size, buttonRef, dataLabel, icon, children, ...elemProps} = this.props;
 
     return (
-      <ButtonBaseCon {...elemProps} innerRef={buttonRef}>
-        {elemProps.icon && <ButtonLabelIcon {...elemProps} />}
-        <ButtonBaseLabel buttonSize={elemProps.buttonSize} buttonType={elemProps.buttonType}>
-          {elemProps.children}
+      <ButtonBaseCon variant={variant} size={size} innerRef={buttonRef} {...elemProps}>
```

Comment:
By default, a `<button>` has a `type` of "submit". Most developers I've encountered assume the default type is "button" which has no special connotation inside of a `<form>` element. Normally this might not matter, but in IE11 if there is no `<form>` element, the document will be treated like a form. At my last place, we set the default `type="button"` to make it so that submit and reset buttons were explicit changes to how a button functions inside a form.

What is our stance on how this should work. The difference is subtle, but profound if type is omitted and "button" is assumed default:

```tsx
<form onSubmit={onSubmit}>
  {/* oops. This is the first "submit" type. If a user hits the enter key in an input, this will be "clicked" and both `onCancel` AND `onSubmit` will be executed. */}
  <Button onClick={onCancel}>Cancel</Button>
  <Button type='submit' variant={Button.Variant.Primary}>Submit</Button>
</form>
```

It is probably fine to fallback to the default in how HTML works, but this is a weird gotcha from HTML's colorful history.

Reply:
Interesting! I've always assumed `type="submit"` by default. I'm not against adding `type="button"`, but it does feel a little weird to override the HTML default

Reply:
Technically React does this with `onChange`. I'm on the fence about this too. There are drawbacks either way. The most notable one I can think of is that if we want to reverse this behavior it'll be a fairly large (and impactful) breaking change. Something to chew on before we decide.

Reply:
Though... if we do end up with a `<SubmitButton>` then yes, we should be defaulting to `type=button`... again just has to be well-documented. Maybe IN ALL CAPS or something ;)

Reply:
So is the consensus to think on this a bit more? I don't see it as a huge blocker for this PR.